compat/mingw.c

```diff
@@ -3587,8 +3587,29 @@ static void setup_windows_environment(void)
 			struct strbuf buf = STRBUF_INIT;
 			strbuf_addstr(&buf, tmp);
 			if ((tmp = getenv("HOMEPATH"))) {
+				/*
+				 * Since Vista, HOMEDRIVE/HOMEPATH are set to Windows\System32
+				 * for secondary logons/services instead of being unset.
+				 * We don't want to (and likely can't) save stuff there
+				 * so ignore them in such a case.
+				 */
+				wchar_t w_system_dir[MAX_PATH];
+				wchar_t *w_buf;
+				size_t len_w_buf;
 				strbuf_addstr(&buf, tmp);
-				if (is_directory(buf.buf))
+				if (!(GetSystemDirectoryW(
+						(LPWSTR)w_system_dir,
+						ARRAY_SIZE(w_system_dir))))
+					die("GetSystemDirectoryW failed: %ld",
+					    GetLastError());
+				len_w_buf = buf.len + 1;
+				w_buf = alloca(len_w_buf * sizeof(wchar_t));
+				xutftowcs(w_buf, buf.buf, len_w_buf);
+
+				/* The defaults are likely set from GetSystemDirectory, too,
+				   so the values should be exactly the same */
+				if ((wcscmp(w_buf, w_system_dir) != 0) &&
```

Comment:
No, we are comparing paths, not strings. Therefore, we have to use the case-insensitive variant of the comparison. Together with my advice to keep handling paths in UTF-8 wherever possible, this would then become `strcasecmp(...)`. Also: in Git's source code, we don't add the `!= 0` in conditionals.

Reply:
No, the goal was not to ignore that path _in any format_. The goal was to at least respect the case-insensitive nature of the paths.

Just because we don't want to fully canonicalize the path doesn't mean that we have to go to the other extreme and _expect_ `HOMEDRIVE`/`HOMEPATH` to match the case of `GetSystemDirectory()`.